core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -188,39 +189,33 @@ private[spark] class CoarseMesosSchedulerBackend(
           taskIdToSlaveId(taskId) = slaveId
           slaveIdsWithExecutors += slaveId
           coresByTaskId(taskId) = cpusToUse
-          val task = MesosTaskInfo.newBuilder()
+          // Gather cpu resources from the available resources and use them in the task.
+          val (remainingResources, taskResources) =
+            partitionResources(offer.getResourcesList, "cpus", cpusToUse)
```

Comment:
these are really cpu resources right? Can we be more explicit about the naming? Also down there we do the same thing for memory. You could rewrite this so it's more readable:
```
val (cpuResourcesRemaining, cpuResourcesToUse) =
  partitionResources(offer.getResourcesList, "cpus", cpusToUse)
val (_, memResourcesToUse) =
  partitionResources(offer.getResourcesList, "mem", MemoryUtils.calculateTotalMemory(sc))
```